### PR TITLE
Fix error message

### PR DESCRIPTION
**Pull request by ivanprado**
It was always empty. 

Also testing for error messages.

### CODE REVIEW

File: tests/test_plan.py [hunk 1]
```diff
@@ -43,6 +44,30 @@ def _final_kwargs_spec(plan):
     return plan[-1][1]
 
 
+@pytest.fixture
+def captured_errors(monkeypatch):
+    """
+    Return a dict that will capture the last errors raised by ``andi.plan``
+    """
```
Comment (kmike) [inline tests/test_plan.py]: I wonder if it makes more sense to store all the structure of the errors in an attribute of NotProvidableError instead.

**Comment by kmike** [comment]
@ivanprado could you please check test failure?

**Comment by ivanprado** [comment]
I have refactored the code to store the structure of the error in the exception. Most of the work was done in commit https://github.com/scrapinghub/andi/pull/14/commits/2dba95e05dd603aed0d203602e688883fc9ce1f2  . I'm not sure I'm happy with the result, it adds more boilerplate, but well, is more robust I guess. 

Also, I found two bugs fixed in https://github.com/scrapinghub/andi/pull/14/commits/b0ba93a1c3e22b2782b0be54a031f27f04e1a5e7

File: andi/errors.py [hunk 2]
```diff
@@ -1,4 +1,66 @@
...
 class NonProvidableError(TypeError):
     """ Raised when a type is not providable """
 
+    def __init__(self, class_or_func, errors_per_argument):
+        self.class_or_func = class_or_func
+        self.errors_per_argument = errors_per_argument
+        super().__init__(_exception_msg(class_or_func, errors_per_argument))
+
+
+CyclicDependencyErrCase = namedtuple("CyclicDependencyErrCase",
```
Comment (ivanprado) [inline andi/errors.py]: @kmike should I declare these as private? I'm not entirely clear regarding when to make public or private.

**Comment by kmike** [comment]
Thanks @ivanprado, looks good to me!

**Comment by ivanprado** [comment]
Thanks @kmike . Can we make a release of `andi`? I can make it if you explain to me how.

**Comment by kmike** [comment]
@ivanprado makes sense. For a release, we need to 

* update Changelog (a separate commit / PR)
* After changelog is merged, run `bumpversion minor --dry-run --verbose`, to check which changes are needed to update the version, and then `bumpversion minor` to actually commit them.
* push version change to github, master branch. Pypi release should be handled by CI.

It'd be awesome if you can help with that! If you do it, would you mind creating a page in wiki with the steps you've followed?

**Comment by ivanprado** [comment]
Sure I'll do it. This is the PR for the CHANGELOG: https://github.com/scrapinghub/andi/pull/15

**Comment by ivanprado** [comment]
Wiki page created https://github.com/scrapinghub/andi/wiki/Release-process and new release done!

**Pull request merged**